### ISSUE TITLE
Invalid channel file - ArrayIndexOutOfBoundsException when importing channel xml

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#1579]
When importing a channel in Mirth Connect 2.0 RC2 that was created with Mirth Connect 2.0 RC1 an error is thrown.

This only happens for certain channels.

Error message:

Invalid channel file. 124 : 124
---- Debugging information ----
message             : 124
cause-exception     : java.lang.ArrayIndexOutOfBoundsException
cause-message       : 124
class               : com.mirth.connect.model.Channel
required-type       : java.lang.String
line number         : 78
-------------------------------
com.thoughtworks.xstream.core.TreeUnmarshaller.convert(TreeUnmarshaller.java:88) ...

Imported Issue. Original Details:
Jira Issue Key: MIRTH-1584
Reporter: nicovn
Created: 2010-10-05T01:21:54.000-0700

**Comment by rbeckman-nextgen** [comment]
Try to download the RC2 again, 2.0.0.5059.  There was an issue in 2.0.05056 with importing old channels that we quickly addressed.

Imported Comment. Original Details:
Author: jacobb
Created: 2010-10-05T11:26:05.000-0700

**Comment by rbeckman-nextgen** [comment]
Let me know if the latest build of RC2 fixes the issue for you.  I tested it with your channel and it worked for me in the latest build.

Imported Comment. Original Details:
Author: jacobb
Created: 2010-10-05T13:40:51.000-0700

**Comment by rbeckman-nextgen** [comment]
Hi Jacob,

You're right ... with version 2.0.0.5059 it works fine !

Thanks a lot!

Best Regards

Nico

Imported Comment. Original Details:
Author: nicovn
Created: 2010-10-06T02:15:44.000-0700

**Comment by rbeckman-nextgen** [comment]
I am using the 2.1.0.5389 version and the ArrayIndexOutOfBoundsException was part of the error I received.  I tried to transfer a file from a file reader to a file writer starting and ending with ASCII as the file type.  I did not set up a transformer, as I did not need the message to be transformed.  However, when deploying the channel, I got several files that gave me the same error message, which was:

Unable to route.org.mule.umo.routing.RoutingException: Failed to transform message before applying the filter. Failed to route event via endpoint: ImmutableMuleEndpoint{connector=com.mirth.connect.connectors.file.FileConnector@79a5d7e0, endpointUri=file://c:/data/lpnt/bbk, transformer=Transformer{name='ByteArrayToString', returnClass=false, returnClass=false, sourceTypes=[class [B, class java.lang.String]}, name='_fileEndpoint#-2125213528', type='receiver', properties={}, transactionConfig=org.mule.impl.MuleTransactionConfig@2090f5c5, filter=null, deleteUnacceptedMessages=false, initialised=true, securityFilter=null, synchronous=true, initialState=started, createConnector=0}. Message payload is of type: java.lang.String

Any ideas on how to remedy this?  Thanks!


Imported Comment. Original Details:
Author: chrismcnatt
Created: 2011-05-18T14:36:50.000-0700
